linode/nb/framework_resource_schema.go

```diff
@@ -0,0 +1,146 @@
+package nb
+
+import (
+	"github.com/hashicorp/terraform-plugin-framework-validators/int64validator"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/int64default"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringdefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+var frameworkResourceSchema = schema.Schema{
+	Version: 1,
+	Attributes: map[string]schema.Attribute{
+		"id": schema.Int64Attribute{
+			Description: "The unique ID of the Linode NodeBalancer.",
+			Computed:    true,
+		},
+		"label": schema.StringAttribute{
+			Description: "The label of the Linode NodeBalancer.",
+			Optional:    true,
+		},
+		"region": schema.StringAttribute{
+			Description:   "The region where this NodeBalancer will be deployed.",
+			Optional:      true,
+			Computed:      true,
+			PlanModifiers: []planmodifier.String{stringplanmodifier.RequiresReplace()},
+			Default:       stringdefault.StaticString("us-east"),
+		},
+		"client_conn_throttle": schema.Int64Attribute{
+			Description: "Throttle connections per second (0-20). Set to 0 (zero) to disable throttling.",
+			Validators: []validator.Int64{
+				int64validator.Between(0, 20),
+			},
+			Optional: true,
+			Computed: true,
+			Default:  int64default.StaticInt64(0),
+		},
+		"hostname": schema.StringAttribute{
+			Description: "This NodeBalancer's hostname, ending with .nodebalancer.linode.com",
+			Computed:    true,
+		},
+		"ipv4": schema.StringAttribute{
+			Description: "The Public IPv4 Address of this NodeBalancer",
+			Computed:    true,
+		},
+		"ipv6": schema.StringAttribute{
+			Description: "The Public IPv6 Address of this NodeBalancer",
+			Computed:    true,
+		},
+		"created": schema.StringAttribute{
+			Description: "When this NodeBalancer was created.",
+			Computed:    true,
+		},
+		"updated": schema.StringAttribute{
+			Description: "When this NodeBalancer was last updated.",
+			Computed:    true,
+		},
```

Comment:
I think we may use rfc 3339 custom types here, not very necessary though, but I think that's a nice to have thing and an additional layer of redundancy.